java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -242,6 +250,37 @@ public String getFileKey() {
       return sourceFile.getKey();
     }
 
+    @CheckForNull
+    private RuleKey getRuleKey(JavaCheck check) {
```

Comment:
@benzonico I fully share, right now the index is per project, it should be at least per file or on the go, ticket created: https://jira.sonarsource.com/browse/SONARJAVA-1256